### PR TITLE
[MULTIARCH-3806, 3807] Add IBM Z and IBM Power support for Agent-based installer

### PR DESCRIPTION
**Pull request by SNiemann15**
Version(s):

Issue:
https://issues.redhat.com/browse/OSDOCS-6413
https://issues.redhat.com/browse/MULTIARCH-3806
https://issues.redhat.com/browse/MULTIARCH-3807

Link to docs preview:

- [Preparing to install with the Agent-based Installer](https://69850--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_with_agent_based_installer/preparing-to-install-with-agent-based-installer)
- [Optional configuration parameters](https://69850--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_with_agent_based_installer/installation-config-parameters-agent#installation-configuration-parameters-optional_installation-config-parameters-agent)

QE review:
- [x] QE has approved this change.

Additional information:
Gdoc https://docs.google.com/document/d/15eJK3VbLfzY1oli7JqjZh2lDCMrj2XMYrPBnsprMWu0/edit

### CODE REVIEW

**Comment by alishaIBM** [comment]
@cs-zhang could you approve/review for IBM Power? Thanks.

**Review by cs-zhang — APPROVED** [review]
/lgtm

**Review by Neeraj8418 — COMMENTED** [review]
Check for the console prompt like we have for x86. Are we going to have the same this for s390x aswell.?
https://console.redhat.com/openshift/install/metal

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (skopacz1) [inline modules/installing-ocp-agent-ibm-z.adoc:8]: If this is only needed for IBM Z users, is there any way you could indicate this condition through either the title or the description? The method I've been using in the [main Agent installation page](https://docs.openshift.com/container-platform/4.14/installing/installing_with_agent_based_installer/installing-with-agent-based-installer.html) is starting the heading with "Optional". It would look like this:

```suggestion
= Optional: Manually adding {ibm-z-title} agents
```

If you do this, the abstract below might need reframing to indicate this procedure as not necessary for regular users but necessary for IBM Z users. Another option would be something more conditional than optional like this:
```suggestion
= {ibm-z-title} only: Manually adding {ibm-z-title} agents
```

In any case I'll leave it up to you, I'd just prefer some indication that this isn't a required part of the workflow for non-IBM Z users.

File: modules/installing-ocp-agent-download.adoc
Comment (skopacz1) [inline modules/installing-ocp-agent-download.adoc:14]: Just checking: Should this note have consistent wording with [this other note](https://github.com/openshift/openshift-docs/pull/69850/files#diff-712d68c5898de472eec19bbf3b4d7e1bda1c35b74926b9af5f0ac5368f3dd22cR14)?

File: modules/installing-ocp-agent-inputs.adoc
Comment (skopacz1) [inline modules/installing-ocp-agent-inputs.adoc:80]: Sorry to add last minute work to your plate, but the [Agent config reference](https://docs.openshift.com/container-platform/4.14/installing/installing_with_agent_based_installer/installation-config-parameters-agent.html) has been merged and is live, and that page has references to valid architectures too. Specifically, the descriptions of [compute.architecture](https://docs.openshift.com/container-platform/4.14/installing/installing_with_agent_based_installer/installation-config-parameters-agent.html#:~:text=compute%3A%0A%20%20architecture%3A-,Determines%20the%20instruction%20set%20architecture%20of%20the%20machines%20in%20the%20pool.%20Currently%2C%20clusters%20with%20varied%20architectures%20are%20not%20supported.%20All%20pools%20must%20specify%20the%20same%20architecture.%20Valid%20values%20are%20amd64%20and%20arm64.,-String) and [controlPlane.architecture](https://docs.openshift.com/container-platform/4.14/installing/installing_with_agent_based_installer/installation-config-parameters-agent.html#:~:text=controlPlane%3A%0A%20%20architecture%3A-,Determines%20the%20instruction%20set%20architecture%20of%20the%20machines%20in%20the%20pool.%20Currently%2C%20clusters%20with%20varied%20architectures%20are%20not%20supported.%20All%20pools%20must%20specify%20the%20same%20architecture.%20Valid%20values%20are%20amd64%20and%20arm64.,-String).


As of 2/9, you can find these Agent specific parameters in lines [492](https://github.com/openshift/openshift-docs/blob/main/modules/installation-configuration-parameters.adoc#L492) and [585](https://github.com/openshift/openshift-docs/blob/main/modules/installation-configuration-parameters.adoc#L585) of modules/installation-configuration-parameters.adoc

**Review by skopacz1 — COMMENTED** [review]
Just left a few comments/questions from an Agent docs perspective, let me know if you have any questions. This looks really good!

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (Neeraj8418) [inline modules/installing-ocp-agent-ibm-z.adoc:32]: Remove either rd.dasd or rd.zfcp. Keep only one here and mention other one in callout.

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (Neeraj8418) [inline modules/installing-ocp-agent-ibm-z.adoc:81]: change the network from  --network bridge=br-ocp,mac=<mac_address> \ 
to 
--network network:macvtap ,mac=<mac_address> \

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (Neeraj8418) [inline modules/installing-ocp-agent-ibm-z.adoc:55]: What is meant by CMS? Is it a guest vm console to zvm?

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (SNiemann15) [reply modules/installing-ocp-agent-ibm-z.adoc:3]: Yes correct, since IBM Z currently only works when through PXE assets I think it better fits into the PXE [assembly](https://github.com/openshift/openshift-docs/pull/69850/files#diff-f54c6c49df16118f8aa38f436a4b946609d0c77741c6bb37835417d583c435a4).

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (SNiemann15) [reply modules/installing-ocp-agent-ibm-z.adoc:8]: For IBM Z customers it's not optional. Therefore, I'll take your second suggestion.

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (SNiemann15) [reply modules/installing-ocp-agent-ibm-z.adoc:55]: Yes see CMS Guide [here](chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://www-40.ibm.com/servers/resourcelink/svc0302a.nsf/pages/zVMV7R2sc246266/$file/dmsb3_v7r2.pdf).

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (SNiemann15) [reply modules/installing-ocp-agent-ibm-z.adoc:32]: I prefer to keep as is to stay in sync with what we have for [SNO](https://docs.openshift.com/container-platform/4.14/installing/installing_sno/install-sno-installing-sno.html#installing-sno-on-ibm-z_install-sno-installing-sno-with-the-assisted-installer)

File: modules/installing-ocp-agent-download.adoc
Comment (SNiemann15) [reply modules/installing-ocp-agent-download.adoc:14]: Different wording is on purpose

**Comment by SNiemann15** [comment]
/label peer-review-needed

SME approvals done, QE approval pending

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z.adoc:15]: I think that this should probably be revised into two separate modules. There are two procedures here. I like this organization really, but I don't think it complies with our one-procedure-per-assembly rule: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#writing-procedures " In most cases, create your procedures as individual modules. . ."

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z.adoc:19]:
```suggestion
. Create a parameter file for the z/VM guest:
```

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z.adoc:38]:
```suggestion
<1> For the `coreos.live.rootfs_url=` artifact, specify the matching `rootfs` artifact for the `kernel` and `initramfs` that you are booting. Only HTTP and HTTPS protocols are supported.
```

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z.adoc:39]: This might need revised for clarity. I'll take a stab at it: 

```suggestion
<2> For the `ip=` parameter, assign the IP address automatically using DHCP, or manually assign the IP address, as described in "Installing a cluster with z/VM on {ibm-z-name} and {ibm-linuxone-name}".
```

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z.adoc:52]:
```suggestion
You can use the CP PUNCH command or, if you use Linux, the **vmur**, command to transfer files between two z/VM guest virtual machines.
```

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z.adoc:57]: Maybe like... "Use the IPL command to bootstrap..." ? 

Or maybe what you have is correct. It's not very telling what IPL is here, but maybe users will already be familiar with this.

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z.adoc:63]: Consider:

```suggestion
See link:https://www.ibm.com/docs/en/zvm/latest?topic=commands-ipl[IPL] in IBM Documentation for more information.
```

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z.adoc:65]: See above comment for adding this in a separate module

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z.adoc:74]: Is `vm_name` a user-defined variable that should be in angle brackets? https://redhat-documentation.github.io/supplementary-style-guide/#user-replaced-values Maybe not :) I see the rest are in angle brackets so I assume this one is correct.

File: modules/installing-ocp-agent-download.adoc
Comment (stevsmit) [reply modules/installing-ocp-agent-download.adoc:14]: I understand that this note and the one below are supposed to have different word, but below you have: (`s390x` architecture) with architecture in parentheses, and here architecture is not in parentheses.

File: modules/understanding-agent-install.adoc
Comment (stevsmit) [inline modules/understanding-agent-install.adoc:14]: See note above about whether architecture should, or should not, be in parentheses.

Comment (SNiemann15) [reply modules/understanding-agent-install.adoc]: Shouldn't

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (SNiemann15) [reply modules/installing-ocp-agent-ibm-z.adoc:74]: Good catch not sure why I didn't change that one has to be <> too.

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z.adoc:16]: Introductory clause: 

```suggestion
Depending on your {ibm-z-name} environment, you can choose from the following options:
```

File: modules/installing-ocp-agent-ibm-z-kvm.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z-kvm.adoc:9]:
```suggestion
Use the following procedure to manually add {ibm-z-name} agents with {op-system-base} KVM.
```

File: modules/installing-ocp-agent-ibm-z-kvm.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z-kvm.adoc:15]: What is `virt-install`? My guess is that it's the command to deploy the virtualization installer tool. So maybe something like

```suggestion
. Deploy the  virtualization installer tool using the following parameters:
```

or maybe

```suggestion
. Enter the following command to deploy the  virtualization installer tool:
```

Or some variation of that. You could even include `virt-install` if you wanted. eg: 

```suggestion
. Using the `virt-install` command to deploy the virtualization installer tool, passing in the following credentials:
```

Just kicking around ideas here.

File: modules/installing-ocp-agent-ibm-z-zvm.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z-zvm.adoc:13]:
```suggestion
. Create a parameter file for the z/VM guest:
```

File: modules/installing-ocp-agent-ibm-z-zvm.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z-zvm.adoc:51]: See other comment about this section if you want to consider revising it according to my suggestion (or not. up to you).

File: modules/installing-ocp-agent-ibm-z-zvm.adoc
Comment (stevsmit) [inline modules/installing-ocp-agent-ibm-z-zvm.adoc:57]: Same here.

File: modules/installing-ocp-agent-ibm-z.adoc
Comment (SNiemann15) [reply modules/installing-ocp-agent-ibm-z.adoc]: yes z/VM users are familiar with that command its z/VM basics.  "to IPL" is evn used as a verb in the z/VM environment. in addition there's a link to the command description.

File: modules/installing-ocp-agent-ibm-z-zvm.adoc
Comment (SNiemann15) [reply modules/installing-ocp-agent-ibm-z-zvm.adoc:51]: I would like to keep it IPL is known to z/VM users

**Comment by Neeraj8418** [comment]
Can we add Disconnected install support for s390x as well like we have for other arch in below doc. This has been recently tested successfully.
[ttps://69850--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_with_agent_based_installer/preparing-to-install-with-agent-based-installer](https://69850--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_with_agent_based_installer/preparing-to-install-with-agent-based-installer)

**Comment by holgwolf** [comment]
LGTM

File: modules/installing-ocp-agent-ibm-z-zvm.adoc
Comment (kcarmichael08) [inline modules/installing-ocp-agent-ibm-z-zvm.adoc:9]:
```suggestion
Use the following procedure to manually add {ibm-z-name} agents with z/VM.
```

**Review by kcarmichael08 — COMMENTED** [review]
I'm not going to hold up merging but consider fixing this the next time you revise it?

**Comment by kcarmichael08** [comment]
One other thing - no version is listed but I'm assuming from the Jira ticket that this is for 4.15 so I'll cherry pick to 4.15. I guess if there are additional versions needed, I could cherry pick later or we can do a manual CP.

**Comment by SNiemann15** [comment]
> One other thing - no version is listed but I'm assuming from the Jira ticket that this is for 4.15 so I'll cherry pick to 4.15. I guess if there are additional versions needed, I could cherry pick later or we can do a manual CP.

Sorry forgot to add it. Yes 4.15 and later is correct. Thanks @kcarmichael08

**Comment by SNiemann15** [comment]
The cherrypick to 4.15 https://github.com/openshift/openshift-docs/pull/71771 did not get merged yet. 
This PR also needs to be cherrypicked to 4.16

File: modules/installing-ocp-agent-ibm-z-zvm.adoc
Comment (SNiemann15) [reply modules/installing-ocp-agent-ibm-z-zvm.adoc:9]: Covered in this PR: https://github.com/openshift/openshift-docs/pull/73204

**Pull request merged**
